S/SuiteSparse/CXSparse/build_tarballs.jl

```diff
@@ -1,38 +1,37 @@
 include("../common.jl")
 
 name = "CXSparse"
+version = v"4.4.0"
+SS_version_str = "7.7.0"
+SS_version = VersionNumber(SS_version_str)
 
-sources = [
-    GitSource("https://github.com/DrTimothyAldenDavis/SuiteSparse.git",
-              "538273cfd53720a10e34a3d80d3779b607e1ac26"),
-]
+sources = suitesparse_sources(SS_version)
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/SuiteSparse
-FLAGS+=(INSTALL="${prefix}" INSTALL_LIB="${libdir}" INSTALL_INCLUDE="${prefix}/include" CFOPENMP=)
-
-if [[ ${target} == *mingw32* ]]; then
-    FLAGS+=(UNAME=Windows)
-    FLAGS+=(LDFLAGS="${LDFLAGS} -L${libdir} -shared")
-else
-    FLAGS+=(UNAME="$(uname)")
-    FLAGS+=(LDFLAGS="${LDFLAGS} -L${libdir}")
+PROJECTS_TO_BUILD="cxsparse"
+CMAKE_OPTIONS+=(
+        -DSUITESPARSE_USE_SYSTEM_SUITESPARSE_CONFIG=ON
+    )
+if [[ "${target}" == aarch64-apple-* ]]; then
+    # Linking libomp requires the function `__divdc3`, which is implemented in
+    # `libclang_rt.osx.a` from LLVM compiler-rt.
+    CMAKE_OPTIONS+=(
+        -DCMAKE_SHARED_LINKER_FLAGS="-L${libdir}/darwin -lclang_rt.osx"
+    )
 fi
+""" * build_script(true)
 
-make -j${nproc} -C CXSparse "${FLAGS[@]}" library CFOPENMP="$CFOPENMP"
-make -j${nproc} -C CXSparse "${FLAGS[@]}" install CFOPENMP="$CFOPENMP"
-
-install_license LICENSE.txt
-"""
-
-platforms = supported_platforms(;experimental=true)
-
-dependencies = [
-    Dependency("SuiteSparse_jll")
-]
+# Add dependency on SuiteSparse_jll
+dependencies = append!(dependencies, [
+    BuildDependency(PackageSpec(name="LLVMCompilerRT_jll", uuid="4e17d02c-6bf5-513e-be62-445f41c75a11", version=v"13.0.1+1");
```

Comment:
For what is worth, with #7877 now merged you can use v16.0.6 (but note that you always need to match the version of the llvm compiler used below).